src/main/java/ca/corefacility/bioinformatics/irida/service/analysis/workspace/galaxy/AnalysisWorkspaceServiceGalaxy.java

```diff
@@ -295,14 +301,18 @@ private void prepareReferenceFile(ReferenceFile referenceFile, History workflowH
 			WorkflowDetails workflowDetails, WorkflowInvocationInputs inputs)
 			throws UploadException, GalaxyDatasetException, WorkflowException {
 
-		Dataset referenceDataset = galaxyHistoriesService.fileToHistory(referenceFile.getFile(), InputFileType.FASTA,
+		IridaTemporaryFile iridaTemporaryFile = iridaFileStorageUtility.getTemporaryFile(referenceFile.getFile());
+
+		Dataset referenceDataset = galaxyHistoriesService.fileToHistory(iridaTemporaryFile.getFile(), InputFileType.FASTA,
 				workflowHistory);
 
 		String workflowReferenceFileInputId = galaxyWorkflowService.getWorkflowInputId(workflowDetails,
 				referenceFileLabel);
 
 		inputs.setInput(workflowReferenceFileInputId,
 				new WorkflowInvocationInputs.WorkflowInvocationInput(referenceDataset.getId(), WorkflowInvocationInputs.InputSourceType.HDA));
+
+		iridaFileStorageUtility.cleanupDownloadedLocalTemporaryFiles(iridaTemporaryFile);
```

Comment:
Should this be in a try/catch or finally block?  That way if something fails, the temp file isn't left there.

Reply:
Yeah that might be a good idea. I will take a look :)

Reply:
Added try/catch to try to delete the file again if an exception is thrown so that the temp file is not left sitting behind in 5579950

Reply:
Updated as discussed in ed329514516fc78562b1070af2fde122f529ae42